### PR TITLE
misc: remove related images/contents/snapshots in cleanup progress 

### PR DESCRIPTION
**Pull request by ChengyuZhu6**
The cleanup process exits before completion. In addition to snapshots, we also need to clean up related content and images.

Depends on Ryan's PR #605.

### CODE REVIEW

**Comment by ChengyuZhu6** [comment]
cc @ryansavino

**Comment by ryansavino** [comment]
I fixed a few things in #605 per review comments. You may want to re-cherrypick the commits.

**Comment by ChengyuZhu6** [comment]
> I fixed a few things in #605 per review comments. You may want to re-cherrypick the commits.

Done.

File: misc/snapshotter/snapshotter.sh [hunk 1]
```diff
@@ -192,36 +192,81 @@ function deploy_snapshotter() {
...
     if [ ! -z "$pid" ]; then
-        local ctr_args=""
-        if [[ " k3s k3s-agent rke2-agent rke2-server " =~ " ${CONTAINER_RUNTIME} " ]]; then
-            ctr_args="--address /run/k3s/containerd/containerd.sock "
-        fi
-        ctr_args+="--namespace k8s.io snapshot --snapshotter nydus"
-        for i in $(nsenter -t 1 -m ctr ${ctr_args} list | grep -v KEY | cut -d' ' -f1); do
-            nsenter -t 1 -m ctr ${ctr_args} rm $i || true
-        done
+        remove_images
```
Comment (ryansavino) [inline misc/snapshotter/snapshotter.sh:252]: @ChengyuZhu6 wondering if this call to cleanup images even needs to be inside the pid check conditional statement. Why not just run it every time?

**Review by ryansavino — APPROVED** [review]

**Review by AdithyaKrishnan — APPROVED** [review]
LGTM

**Comment by imeoer** [comment]
LGTM

File: misc/snapshotter/snapshotter.sh (hunk 1 above)
Comment (ChengyuZhu6) [reply misc/snapshotter/snapshotter.sh:252]: If the snapshotter isn’t running, the `remove_images` function won’t have any effect.

**Pull request merged**